pallets/liquidity-pools/src/lib.rs

```diff
@@ -689,6 +692,26 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::weight(10_000)]
+		#[pallet::call_index(10)]
+		pub fn schedule_upgrade(origin: OriginFor<T>, contract: DomainAddress) -> DispatchResult {
+			ensure_root(origin)?;
+			// get account id of this pallet
+			let who = T::PalletId::get().into_account_truncating();
```

Comment:
If we start using system internal accounts, we maybe think about adapting the `T::OutboundQueue` implementation to use something like 

```rust
enum Account<AccountId> {
   Interal,
   External(AccountId)
}
```

and let the queue handle paying for fees, I.e. take them straight from the treasury or so. Otherwise, we must fund the individuel pallets with unknown high amounts, which will be insane overhead in the long run.

Reply:
@mustermeiszer that's also what I was wondering in a thread above but note that this extrinsic can only be called by the `Origin::Root` so it's not the pallet account that will be paying for this extrinsic, we just use the pallet-derived-account as the `who` for `OutboundQueue::submit`.

Reply:
But the outbound queue makes use of the account and triggers (possibly) ethereum tx afterwards. And the EVM will use the given account for fee paying. IIRC.

Reply:
@mustermeiszer thanks for the deeper insight! I've written a basic test to cover the `EthereumXcmRouter` case and it fails due to "UnableToWithdrawFunds`, despite me funding the `LiquidityPools` pallet account, which is unexpected. https://github.com/centrifuge/centrifuge-chain/pull/1469/commits/6804dc1a502a3b7b7108c183b06a20c8a4af8b2c

In any case, we do have to look further into this since it doesn't work as is, at least not trivially (unless if I am making some weird mistake in that test)

cc @AStox

Reply:
Update: I was indeed making a mistake. It's now fixed. I was minting CFG instead of GLMR to the `LiquidityPools` account 🙉 

@mustermeiszer on your remark:

> we must fund the individuel pallets with unknown high amounts, which will be insane overhead in the long run.

Note that this is the only pallet and the only edge case where this scenario applies. And we don't plan on running `schedule_upgrade` every day, so I don't think this is a problem we actually have to solve.

Reply:
@mustermeiszer or we drop `PalletId` in favour of `TreasuryAccount` and use that as the `who` passed down to `Outbound::queue`. wdyt?

Reply:
Yes, that I think is a great idea! it is anyways always democracy calling that, so the network should also pay!

Reply:
Sweet, that makes a lot of sense. @AStox sorry buddy! We can just rename the new `PalletId` type param you added to the `Config` to `type TreasuryAccount: Get<Self::AccountId>;` and then pass `T::TreasuryAccount::get()` as the `who` of `submit` 🙏